### PR TITLE
[Up-477] Optimization example - remove unused resources

### PR DESCRIPTION
**Pull request by kellemNegasi**

### CODE REVIEW

**Review by gunnsth — CHANGES_REQUESTED** [review]
Can you update the README also under compress/ folder?

**Comment by gunnsth** [comment]
Maybe we should mention which filters are lossy vs lossless.  Removing unused resources is lossless for the most parts.

**Comment by gunnsth** [comment]
@kellemNegasi can you update this and the other PR?  A new release of unipdf has just been published so this should be ready once updated.

**Comment by kellemNegasi** [comment]
> @kellemNegasi can you update this and the other PR?  A new release of unipdf has just been published so this should be ready once updated.

Ok. I will do.

**Comment by kellemNegasi** [comment]
> Maybe we should mention which filters are lossy vs lossless. Removing unused resources is lossless for the most parts.

@gunnsth I have tried to mention which filters are lossless on the README file. Can you check and see if I have to add or remove some filters to or from the list.

File: compress/README.md [hunk 1]
```diff
@@ -14,11 +14,14 @@ type Options struct {
 	CleanFonts                      bool
 	SubsetFonts                     bool
 	CleanContentstream              bool
+	CleanUnusedResources            bool
 }
 ```
 
+From the available filters `CleanFonts`, `CleanUnusedResources`, `CombineDuplicateStreams`, `CombineDuplicateDirectObjects`,`CombineDuplicateDirectObjects`, `CleanFonts` and `CompressStreams` enable lossless compressions.
```
Comment (gunnsth) [inline compress/README.md:21]: UseObjectStreams also
CombineIdenticalIndirectObjects also

Actually most of them are lossless except ImageUpperPPI and ImageQuality.

**Review by gunnsth — APPROVED** [review]
LGTM

**Pull request merged**